pages/menu/edit-customer/[id].js

```diff
@@ -0,0 +1,57 @@
+import Header from "../../../components/Header/Header";
+import { Form, FormButton } from "../../../pagestyles/styles";
+import { useRouter } from "next/router";
+import useSWR from "swr";
+
+export default function EditCustomer() {
+  const router = useRouter();
+  const { id } = router.query;
+  const { mutate } = useSWR("/api");
+
+  async function handleEditSubmit(e, id) {
+    e.preventDefault();
+    const formData = new FormData(e.target);
+    const customer = Object.fromEntries(formData);
+    console.log(id);
+    const response = await fetch(`/api/${id}`, {
+      method: "PATCH",
+      body: JSON.stringify(customer),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+
+    if (!response.ok) {
+      console.error(response.status);
+
+      return;
+    }
+    mutate();
+
+    e.target.reset();
+    router.push("/");
+  }
+  return (
+    <>
+      <Header />
+      <Form onSubmit={(e) => handleEditSubmit(e, id)}>
```

Comment:
```suggestion
      <Form onSubmit={handleEditSubmit}>
```
It's enough here to pass your function to the `onSubmit` like this. The other way is not incorrect and works fine too, but if you're just forwarding the event to another function without doing anything else in the arrow function, it's more clean to directly reference the function as shown above.

Reply:
but in this case i still need the id ? :)

Reply:
Johannes, I think in this case you are right. Since ``Form`` is not a React-Component you want to call the function here.